portfolio/src/main/webapp/script.js

```diff
@@ -35,29 +35,38 @@ function addRandomFact() {
   factContainer.innerText = fact; 
 }
 
-const size = 30;
-var canvas;
-var ctx;
+const SNAKE_SQUARE_SIZE = 30;
 var x = new Array(5);
 var y = new Array(5);
 
+function getCanvasAndContext() {
+  var canvas = document.getElementById('snakeCanvas'); // Get the canvas element by Id
+  var ctx = canvas.getContext('2d'); // Canvas 2d rendering context
+  return {
+    canvas: canvas,
+    ctx: ctx,
+  };
+}
+
 function initSnake() {
-  canvas = document.getElementById('snakeCanvas'); // Get the canvas element by Id
-  ctx = canvas.getContext('2d'); // Canvas 2d rendering context
-  for (var z=0; z < 5; z++) {
+  for (var z = 0; z < 5; z++) {
     x[z] = 10;
     y[z] = 10;
   }
   drawSnake();
 }
 	
 function drawSnake() {
+  var values = getCanvasAndContext();
+  var canvas = values.canvas;
+  var ctx = values.ctx;
+
   ctx.clearRect(0, 0, canvas.width, canvas.height);
   
-  for (var z=0; z<5; z++) {
+  for (var z = 0; z < 5; z++) {
```

Comment:
this could also be a constant?

Reply:
I turned the value 5 into a constant called SNAKE_LENGTH in another pull request.